### ISSUE TITLE
Wrong body shift and disappearing of the vertical scrollbar after Alert

### ISSUE THREAD

**Issue by utilmind** [MohammadYounes/AlertifyJS#206]
Hi! I have noticed a bug (or bug-like feature). Vertical scroll disappers and there is no possibility to vertically scroll the document anymore after the alertify WHEN the document scrolled to bottom (to the end point of scrollbar).

This issue occurs only when preventBodyShift set to TRUE. (I like this feature and don't want to disable it.) Issue noticed both in Chrome and FireFox.

Following page temporaray reproduces the error: https://calculium.com/ -- scroll to the bottom and click "Test alert" in the footer. (I will remove that link in 2 weeks after this post. This is temporary demo page.)

But I have managed to fix this issue. The solutions is to decrement the vertical scroll position to 1 pixel upon restoring previous scroll position.

So when we save the scroll position, we should decrease it to 1 pixel in case if the page vertically scrolled down to the bottom:
```
var scrollX, scrollY;
function saveScrollPosition(){
    scrollX = getScrollLeft();
    scrollY = getScrollTop();

    var scrollHeight = (document.documentElement && document.documentElement.scrollHeight) || document.body.scrollHeight,
        viewportHeight = window.innerHeight || (document.documentElement && document.documentElement.clientHeight) || document.body.clientHeight;
    if (scrollY >= scrollHeight-viewportHeight)
      --scrollY;
}
```

I would appreciate if this fx appear in new versions of Alertify or maybe you know even better solution of this issue, without shifting the page up to 1 pixel when it's in bottom.

**Comment by MohammadYounes** [comment]
Hi, Thanks for reporting this!

I checked the sample page and this seems to be related to having `html { height:100%}`, which would cause the `documentElement.scrollHeight` to be equal to `documentElement.clientHeight`, which in turn prevents alertify from [restoring](https://github.com/MohammadYounes/AlertifyJS/blob/master/src/js/dialog/intro.js#L332) the scrollbar.

At the moment I don't really remember why the internal `preventBodyShift` function is wrapped in such condition ... I need to double check before making any change. However, for the time being, you can get rid of the issue by adding `html{height:auto;}` to your page.

**Comment by utilmind** [comment]
Thanks for response! Yes, I have tested and seen that `html { height: 100% }` fixes the problem.

However, unfortunately, I can't use it and really need 100% height for `html` and `body` tags. My website contains some pages without vertical scrollbar, with 100% height, where all content fits the viewport. Example: https://calculium.com/distance/.

I'm not worrying, my fix mentioned above completely satisfies me. But it would be great to fix this problem completely in standard Alertify release. Thank you!

**Comment by MohammadYounes** [comment]
Fixed in a776efb2953592174bed1b89bfbc802e9214cbab
